packages/twenty-server/src/engine/core-modules/auth/services/microsoft-apis.service.ts

```diff
@@ -0,0 +1,218 @@
+      );
+
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
+        workspaceId,
+        'messageChannel',
+      );
+
+    const workspaceDataSource =
+      await this.twentyORMGlobalManager.getDataSourceForWorkspace(workspaceId);
+
+    const scopes = getMicrosoftApisOauthScopes();
+
+    await workspaceDataSource.transaction(async (manager: EntityManager) => {
+      if (!existingAccountId) {
+        await connectedAccountRepository.save(
+          {
+            id: newOrExistingConnectedAccountId,
+            handle,
+            provider: ConnectedAccountProvider.MICROSOFT,
+            accessToken: input.accessToken,
+            refreshToken: input.refreshToken,
+            accountOwnerId: workspaceMemberId,
+            scopes,
+          },
+          {},
+          manager,
+        );
+
+        await messageChannelRepository.save(
+          {
+            id: v4(),
+            connectedAccountId: newOrExistingConnectedAccountId,
+            type: MessageChannelType.EMAIL,
+            handle,
+            visibility:
+              messageVisibility || MessageChannelVisibility.SHARE_EVERYTHING,
+            syncStatus: MessageChannelSyncStatus.ONGOING,
+          },
+          {},
+          manager,
+        );
+
+        await calendarChannelRepository.save(
+          {
+            id: v4(),
+            connectedAccountId: newOrExistingConnectedAccountId,
+            handle,
+            visibility:
+              calendarVisibility || CalendarChannelVisibility.SHARE_EVERYTHING,
+          },
+          {},
+          manager,
+        );
+      } else {
+        await connectedAccountRepository.update(
+          {
+            id: newOrExistingConnectedAccountId,
+          },
+          {
+            accessToken: input.accessToken,
+            refreshToken: input.refreshToken,
+            scopes,
+          },
+          manager,
+        );
+
+        const workspaceMemberRepository =
+          await this.twentyORMGlobalManager.getRepositoryForWorkspace<WorkspaceMemberWorkspaceEntity>(
+            workspaceId,
+            'workspaceMember',
+          );
+
+        const workspaceMember = await workspaceMemberRepository.findOneOrFail({
+          where: { id: workspaceMemberId },
+        });
+
+        const userId = workspaceMember.userId;
+
+        await this.accountsToReconnectService.removeAccountToReconnect(
+          userId,
+          workspaceId,
+          newOrExistingConnectedAccountId,
+        );
+
+        await messageChannelRepository.update(
+          {
+            connectedAccountId: newOrExistingConnectedAccountId,
+          },
+          {
+            syncStage: MessageChannelSyncStage.FULL_MESSAGE_LIST_FETCH_PENDING,
+            syncStatus: null,
+            syncCursor: '',
+            syncStageStartedAt: null,
+          },
+          manager,
+        );
+      }
+    });
```

Comment:
Maybe since this part is not yet implemented, we should put the messageChannel in the `FAILED` syncStage state and put the syncStatus as `NOT_SYNCED`